pkg/client/client.go

```diff
@@ -293,13 +293,13 @@ func isNewDynatraceTokenFormat(token string) bool {
 func (d *DynatraceClient) UpsertSettings(obj SettingsObject) (DynatraceEntity, error) {
 	externalId := idutils.GenerateExternalID(obj.SchemaId, obj.Id)
 
-	// list all settings with matching external ID
+	//list all settings with matching origin object ID
 	settings, err := d.ListSettings(obj.SchemaId, ListSettingsOptions{
 		// we don't care about the actual value setting
 		DiscardValue: true,
 		// only consider objects with an external id that matches the expected one
 		Filter: func(o DownloadSettingsObject) bool {
-			return o.ExternalId == externalId
+			return o.ObjectId == obj.OriginObjectId
```

Comment:
wait, so we're fulling letting the Settings framework handle the "upsert" logic? 

does making this call even make sense anymore then? If it's just used for the old version + existing object check maybe we can just implement a GET by object id rather than iterating?

Reply:
yes, it's completely handled by the settings fw.
yes, actually a get would be also possible. didn't thought about that :+1:

Reply:
As is we load and iterate over potentially hundreds of settings just to find a specific one. 
I would really suggest we replace it with just a get by objectId

Reply:
just saw that the current implementation of `DynatraceClient` does not have a method to Get a single object. Are you OK with doing this "optimization" in a follow-up ? :)